plugins/bigquery/dbt/adapters/bigquery/impl.py

```diff
@@ -84,6 +85,11 @@ def list_schemas(self, database):
                                                 include_all=True)
             return [ds.dataset_id for ds in all_datasets]
 
+    @available
```

Comment:
does this mean that we'll need to do something similar for the Presto and Spark plugins?

Reply:
This `@available` exists to make the relation available to bq's macros so it can implement this macro, because on bigquery that functionality is implemented via the connection.

Spark is a `SQLAdapter` and so it already has the macro available. Though it does look like it's [incomplete](https://github.com/fishtown-analytics/dbt-spark/blob/master/dbt/include/spark/macros/adapters.sql#L76)?

We currently don't support archives on presto, so we don't have to worry about that at all.

Reply:
hah, dang. Ok, thanks for the info, we can handle Spark separately from this for sure